### PR TITLE
Support ephemeral_storage_local_ssd_config

### PR DESCRIPTION
**Pull request by dannawang0221**
Support for GKE ephemeral storage local ssd and local nvme ssd block
fix: https://github.com/hashicorp/terraform-provider-google/issues/14021 

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added support for `ephemeral_storage_local_ssd_config` to `google_container_cluster.node_config`, `google_container_cluster.node_pools.node_config`, `google_container_node_pool.node_config`
```

### CODE REVIEW

**Review by rileykarson — COMMENTED** [review]
See https://github.com/hashicorp/terraform-provider-google/issues/13829, https://github.com/hashicorp/terraform-provider-google/issues/8461 - both of these are supported.

One of the two is only at beta though, so you may want to GA it if it's available in the GA API now.

**Comment by dannawang0221** [comment]
> See [hashicorp/terraform-provider-google#13829](https://github.com/hashicorp/terraform-provider-google/issues/13829), [hashicorp/terraform-provider-google#8461](https://github.com/hashicorp/terraform-provider-google/issues/8461) - both of these are supported.
> 
> One of the two is only at beta though, so you may want to GA it if it's available in the GA API now.
ephemeral_storage_local_ssd_config is different from ephemeral_storage_config. I'll remove local_nvme_ssd_block_config, thanks for reminding.

**Comment by rileykarson** [comment]
Hmm- I see in the docs that "This field is functionally equivalent to the ephemeralStorageConfig" on `ephemeralStorageLocalSsdConfig`. I'm not sure it makes sense to support both as the fields will likely fight over which controls the setting.

**Comment by dannawang0221** [comment]
We won't graduate ephemeralStorageConfig from beta to GA. ephemeralStorageLocalSsdConfig is the GA version for the feature.

**Comment by rileykarson** [comment]
Got it! We likely want to remap `ephemeral_storage_config` on to ephemeralStorageLocalSsdConfig given that- supporting both will cause issues for users who use `ephemeral_storage_config` already, and require handling that'll degrade the behaviour of both fields. It's a strict requirement that the beta Terraform provider is a strict superset of the GA one, and both surfaces are stable.

**Comment by dannawang0221** [comment]
ok, so I should replace ```ephemeralStorageConfig``` with ```ephemeralStorageLocalSsdConfig``` and graduate ```ephemeral_storage_config``` from beta to ga?

**Comment by rileykarson** [comment]
Ah! No, we'd keep the field name `ephemeral_storage_config`, GAing that in Terraform, and remap the implementation of the field to use the new API field.

**Comment by dannawang0221** [comment]
It will be a breaking change for the users who are using ```ephemeralStorageConfig``` currently? If they do a ```get```, they won't be able to find the information since we don't have a backfill from ```ephemeralStorageConfig``` to ```ephemeralStorageLocalSsdConfig```

File: mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 1]
```diff
@@ -745,6 +745,13 @@ ephemeral_storage_config {
   local_ssd_count = 2
 }
 ```
+* `ephemeral_storage_local_ssd_config` - (Optional) Parameters for the ephemeral storage filesystem. If unspecified, ephemeral storage is backed by the boot disk. Structure is [documented below](#nested_ephemeral_storage_local_ssd_config).
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown:748]: nit: Other than the "Structure is documented below." bit here, do you mind copying the descriptions for these fields into the `Description` value on the resource exactly? That's used by Config Connector to generate docs.

**Review by rileykarson — COMMENTED** [review]
Discussed offline, the new field replaces the old beta-only one but they're incompatible w/ no backfill. Given that, we can't remap the field and introducing the new field is correct.

We may want to mark [ephemeral_storage_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#ephemeral_storage_config) as deprecated to push users to use the new field instead. We'll need to outline how to migrate between them in the deprecation message- that's going to be recreating the affected resources, given both fields are immutable.

Terraform deprecations don't fall under GCP policies for this kind of stuff. The ecosystem expectation is that we document the path end users take to resolve the deprecation is all. In the next major release, deprecated fields can be removed.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**